frontend/catalyst/pennylane_extensions.py

```diff
@@ -254,27 +254,49 @@ def _make_jaxpr_check_differentiable(f: Differentiable, grad_params: GradParams,
                 f"results, got '{res.dtype}' at position {pos}."
             )
 
-    _check_created_jaxpr_gradient_methods(f, method, jaxpr)
+    _verify_differentiable_child_qnodes(jaxpr, method)
 
     return jaxpr
 
 
+def _verify_differentiable_child_qnodes(jaxpr, method):
+    """Traverse QNodes being differentiated in the 'call graph' of the JAXPR to verify them."""
+    visited = set()
+
+    def traverse_children(jaxpr):
+        for eqn in jaxpr.eqns:
+            fn = eqn.params.get("fn")
+            if fn and fn not in visited:
+                child = eqn.params.get("call_jaxpr", None)
+                if isinstance(fn, (qml.QNode, Grad)):
+                    _check_created_jaxpr_gradient_methods(fn, method, child)
+                if child and child not in visited:
+                    traverse_children(child)
+            visited.add(fn)
```

Comment:
I doubt it would be, there are other places in the code that access the primitives' parameters by name. I don't know if there's a better way to interface with JAXPR primitives.